Make ASCIILiteral::hash() constexpr. NFCI.

mikhailramalho opened the pull request:
#### d9a9d09eec9ce68a533f6f02de0f2aa1627e017f
<pre>
Make ASCIILiteral::hash() constexpr. NFCI.
<a href="https://bugs.webkit.org/show_bug.cgi?id=297782">https://bugs.webkit.org/show_bug.cgi?id=297782</a>

Reviewed by Yusuke Suzuki.

This PR fixes a compilation issue when building jsc for linux arm32 and amr64
in debug mode:

PCToCodeOriginMap.h: In static member function ‘static constexpr uint64_t JSC::Wasm::OMGOrigin::expectedMagicValue()’:
ValidatedReinterpretCast.h:60:38: error: call to non-‘constexpr’ function ‘unsigned int WTF::ASCIILiteral::hash() const’
   60 |         return ASCIILiteral(fqn).hash(); \
      |                ~~~~~~~~~~~~~~~~~~~~~~^~

* Source/WTF/wtf/text/ASCIILiteral.h:
(WTF::ASCIILiteral::hash const):

Canonical link: <a href="https://commits.webkit.org/299067@main">https://commits.webkit.org/299067@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/952f5498f00139abd815ff13c0f8dd11bcacee81

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/117673 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/37351 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/27978 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/123807 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/69684 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/38043 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/45933 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/89313 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/50349 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/120625 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30284 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/105510 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/69804 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/29347 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/23626 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/67463 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/109783 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/99701 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/23806 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/126900 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/116182 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/44576 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/33547 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/97968 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/44934 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/101737 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/97755 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24876 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43132 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/21093 "Passed tests") | [❌ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/40940 "Hash 952f5498 for PR 49769 does not build (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/44447 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/50122 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/144880 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/43906 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/37279 "Found 1 new JSC binary failure: testapi, Found 20009 new JSC stress test failures: ChakraCore.yaml/ChakraCore/test/Array/array_slice.js.default, ChakraCore.yaml/ChakraCore/test/Array/array_sort2.js.default, ChakraCore.yaml/ChakraCore/test/Array/push2.js.default, ChakraCore.yaml/ChakraCore/test/Array/reverse1.js.default, ChakraCore.yaml/ChakraCore/test/Array/toLocaleString.js.default, ChakraCore.yaml/ChakraCore/test/Date/date_cache_bug.js.default, ChakraCore.yaml/ChakraCore/test/Error/CallNonFunction.js.default, ChakraCore.yaml/ChakraCore/test/Error/stack2.js.default, ChakraCore.yaml/ChakraCore/test/Function/FuncBody.bug227901.js.default, ChakraCore.yaml/ChakraCore/test/Function/apply.js.default ... (failure)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/47253 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/45597 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/952f5498f00139abd815ff13c0f8dd11bcacee81. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49769

mikhailramalho committed:
Make ASCIILiteral::hash() constexpr. NFCI.
https://bugs.webkit.org/show_bug.cgi?id=297782

Reviewed by Yusuke Suzuki.

This PR fixes a compilation issue when building jsc for linux arm32 and amr64
in debug mode:

PCToCodeOriginMap.h: In static member function ‘static constexpr uint64_t JSC::Wasm::OMGOrigin::expectedMagicValue()’:
ValidatedReinterpretCast.h:60:38: error: call to non-‘constexpr’ function ‘unsigned int WTF::ASCIILiteral::hash() const’
   60 |         return ASCIILiteral(fqn).hash(); \
      |                ~~~~~~~~~~~~~~~~~~~~~~^~

* Source/WTF/wtf/text/ASCIILiteral.h:
(WTF::ASCIILiteral::hash const):

Canonical link: https://commits.webkit.org/299067@main

webkit-commit-queue commented:
Committed 299067@main (d9a9d09eec9c): <https://commits.webkit.org/299067@main>

Reviewed commits have been landed. Closing PR #49769 and removing active labels.
